addons/l10n_ar_website_sale/controllers/main.py

```diff
@@ -57,3 +57,18 @@ def checkout_form_validate(self, mode, all_form_values, data):
                     error_message.append(_('For the selected AFIP Responsibility you will need to set CUIT Identification Type'))
 
         return error, error_message
+
+    @http.route()
+    def address(self, **kw):
+        """ Override to reactivate the address_b2b view if not active
+        and the website company country is Argentina. """
+        res = super().address(**kw)
+        website = request.website.sudo()
+        if website.company_id.country_id.code == "AR":
+            address_b2b_view = request.env['ir.ui.view'].with_context(active_test=False).sudo().search([
+                ('key', '=', 'website_sale.address_b2b'),
+                ('website_id', '=', website.id),
+            ])
+            if address_b2b_view and not address_b2b_view.active:
+                address_b2b_view.write({'active': True})
+        return res
```

Comment:
```suggestion
```
seems overkill to me

Reply:
Note that without this, broken DB would remain.. broken.
See https://github.com/odoo/odoo/pull/98084#issuecomment-1225378899 which is still true.
Not saying this should absolutely remain but just to be sure everyone understands that a db admin could break his checkout without noticing it (you disable the settings and are happy, but actually your customer have checkout crashing).

While the constraint will prevent DB to be broken, if a DB is already broken before this fix, it will remain broken (and possibly "silently broken" -> the admin doesn't event know).

I agree this is a bit overkill, (even if I'd do it, as this is encapsulated in the l10n_ar overide), but just to be sure this was correctly figured.

I guess we could agree that there shouldn't be much DB actually broken before this FIX that would really need this part of the patch tho.

At the end, I'll let you decide what you prefer as this is your module, I just suggested the changes ;-)

Reply:
@rdeodoo I would be guessing that not a lot of people would have done that and if they would have the issue, they would already have seen it.  At least for new websites, the problem won't be there.  @jjscarafia @zaoral

Reply:
It can be done like this, but I still think it is a bit overkill.  Up to you @rdeodoo

Reply:
As I said:

> I agree this is a bit overkill, (even if I'd do it, as this is encapsulated in the l10n_ar overide), but just to be sure this was correctly figured.
> I guess we could agree that there shouldn't be much DB actually broken before this FIX that would really need this part of the patch tho.
> At the end, I'll let you decide what you prefer as this is your module, I just suggested the changes ;-)

Let's forget about it :+1: :)